pex/bin/pex.py

```diff
@@ -732,6 +741,9 @@ def main(args=None):
       tmp_name = options.pex_name + '~'
       safe_delete(tmp_name)
       pex_builder.build(tmp_name)
+      if options.validate_ep:
```

Comment:
You could be more uniform, more correct and fail faster with:
```python
pex_builder = build_pex(reqs, options, resolver_options_builder)
pex_builder.freeze()
pex = PEX(pex_builder.path(), interpreter=pex_builder.interpreter, verify_entry_point=options.validate_ep)

if options.pex_name is not None:
  ...
```
As things stand, in the options.pex_name branch, the pex is zipped up before the entrypoint validation which could be wasted work. Worse though is that the builder's custom interpreter, is any, is not used. This could be a problem if the current running interpreter does not match what the built pex needs and syntax differs (a 2/3 split).

Reply:
Moved things up to fail fast.